pyanaconda/modules/subscription/installation.py

```diff
@@ -15,32 +15,446 @@
 from pyanaconda.modules.subscription import system_purpose
 
+from pyanaconda.anaconda_loggers import get_module_logger
+log = get_module_logger(__name__)
+
+
+def start_rhsm_private_bus():
+    """Start the RHSM private DBus.
+
+    The RHSM private DBus session is used for secure credential passing.
+
+    return: address of the RHSM private DBus session
+    rtype: str
+    """
+    rhsm_proxy = RHSM.get_proxy()
+    register_server_result = rhsm_proxy.RegisterServer.Start()
+    return register_server_result
+
+
+class RHSMPrivateBusConnection(Connection):
+    """Representation of a RHSM private bus connection."""
+
+    def __init__(self, rhsm_private_bus_address):
+        super().__init__()
+        self._rhsm_private_bus_address = rhsm_private_bus_address
+
+    def get_new_connection(self):
+        """Connect to the RHSM private bus."""
+        log.info("Connecting to RHSM private bus at %s.", self._rhsm_private_bus_address)
+        connection = pydbus.bus.Gio.DBusConnection.new_for_address_sync(self._rhsm_private_bus_address,
+                                                                        pydbus.bus.Gio.DBusConnectionFlags.AUTHENTICATION_CLIENT,
+                                                                        None, None)
+        connection.pydbus.autoclose = False
+        return connection.pydbus
+
+
 class SystemPurposeConfigurationTask(Task):
     """Installation task for setting system purpose."""
 
     def __init__(self, sysroot, role, sla, usage, addons):
         """Create a new task.
-
-        :param sysroot: a path to the root of the installed system
-        :param lang: a value for LANG locale variable
+        :param str sysroot: a path to the root of the installed system
         """
         super().__init__()
         self._sysroot = sysroot
         self._role = role
         self._sla = sla
         self._usage = usage
         self._addons = addons
+        self.succeeded = Signal()
 
     @property
     def name(self):
         return "Set system purpose"
 
     def run(self):
         system_purpose.give_the_system_purpose(self._sysroot,
-                                                 self._role,
-                                                 self._sla,
-                                                 self._usage,
-                                                 self._addons)
+                                               self._role,
+                                               self._sla,
+                                               self._usage,
+                                               self._addons)
+        self._done()
+
+    @async_action_nowait
+    def _done(self):
+        # tell nayone listening that we are done
+        self.succeeded.emit()
+
+
+class StartRHSMTask(Task):
+    """Start the RHSM DBUS service.
+
+    TODO: this might not be needed once the RHSM DBUS services
+          is a regular part of the installation environment
```

Comment:
This was needed mostly for updates image based development, but I now have custom boot ISOs, so it should be now possible to verify if this is needed in practice. Still, I think we still can drop this only in the final review round or else this will simply not work if an updates image is used.

Reply:
Actually, it looks like the service will get started automatically even with an updates image, likely due to auto-activation. :) So looks like we should be able to drop this completely right away.